File: webpage_snapshot.py
import sys
import imgkit

# Get the URL from the command line arguments
url = sys.argv[1]

# Configure options for imgkit
options = {
    "format": "png",
    "encoding": "UTF-8"
}

# Create a screenshot of the webpage
imgkit.from_url(url, "/home/chatgpt/custom_utilities/utility_library/tmp/webpage_snapshot.png", options=options)

File: webpage_snapshot.js
const puppeteer = require("puppeteer");
const fs = require("fs");
const url = process.argv[2];

(async () => {
    const browser = await puppeteer.launch({args: ["--no-sandbox"], headless: "new"});
    const page = await browser.newPage();
    await page.goto(url);
    await page.screenshot({path: "/home/chatgpt/custom_utilities/utility_library/tmp/screenshot.png"});
    await browser.close();
})();
